### PR TITLE
Added GuestLightBox component and added guest chat feature

### PR DESCRIPTION
**Pull request by abhishekbadoni-bigstep**
- Added GuestLightBox component
- Added compatibility for guest user chat for open conversations

### CODE REVIEW

File: src/components/ConversationWindow.js [hunk 1]
```diff
@@ -273,6 +278,15 @@ class ConversationWindow extends PureComponent {
     this._onTextMessageChanged(event.target.value);
   }
 
+  toggleGuestLightBox() {
+    this.setState({guestLightBoxOpened: !this.state.guestLightBoxOpened})
+  }
+  
+  async onGuestUpdate(guest) {
```
Comment (balramgoyal) [inline src/components/ConversationWindow.js]: `onGuestUpdate` => `onJoinedAsGuest`.  `onJoinedAsGuest` is looking more readable and we have to change everywhere.

File: src/components/ConversationWindow.js [hunk 2]
```diff
@@ -349,9 +363,17 @@ class ConversationWindow extends PureComponent {
   }
   
   sendMessage() {
-    const { conversation, client, userId } = this.props;
+    const { conversation, client, userId, allowGuestUsers } = this.props;
+
+    if (allowGuestUsers && client.isAnonymousUser()) {
+      return this.toggleGuestLightBox();
```
Comment (balramgoyal) [inline src/components/ConversationWindow.js]: `toggleGuestLightBox ` => `toggleGuestJoinForm`.

File: src/components/ConversationWindow.js [hunk 3]
```diff
@@ -644,20 +667,23 @@ class ConversationWindow extends PureComponent {
...
+                  </button>
+
         			</div>
             }
           </React.Fragment>
         }
 
         { this.state.openMediaFile && <MediaLightBox file={this.state.openMediaFile} onCloseClick={()=> this.viewMediaToggle(null)} /> }
+
+        { this.state.guestLightBoxOpened && <GuestLightBox onCloseClick={this.toggleGuestLightBox} onGuestUpdate={this.onGuestUpdate} /> }
```
Comment (balramgoyal) [inline src/components/ConversationWindow.js]: `GuestLightBox ` => `GuestJoinForm`

File: src/components/ConversationWindow.js (hunk 1 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationWindow.js]: Done!

File: src/components/ConversationWindow.js (hunk 2 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationWindow.js]: Done!

File: src/components/ConversationWindow.js (hunk 3 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationWindow.js]: Done!

**Pull request merged**